modules/code-editor/src/views/full/store/index.ts

```diff
@@ -0,0 +1,169 @@
+import { action, observable, runInAction } from 'mobx'
+import path from 'path'
+
+import { EditableFile, FilesDS, FileType } from '../../../backend/typings'
+import { Config, FileFilters, StudioConnector } from '../typings'
+import { FILENAME_REGEX, toastSuccess } from '../utils'
+import { baseAction, baseHook } from '../utils/templates'
+
+import CodeEditorApi from './api'
+import { EditorStore } from './editor'
+
+/** Includes the partial definitions of all classes */
+export type StoreDef = Partial<RootStore> & Partial<Config> & Partial<EditorStore>
+
+class RootStore {
+  public api: CodeEditorApi
+  public editor: EditorStore
+
+  public config: Config
+  public typings: { [fileName: string]: string } = {}
+
+  @observable
+  public files: FilesDS
+
+  @observable
+  public filters: FileFilters
+
+  @observable
+  public fileFilter: string
+
+  constructor({ bp }) {
+    this.api = new CodeEditorApi(bp.axios)
+    this.editor = new EditorStore(this)
+    // Object required for the observer to be useful.
+    this.filters = {
+      filename: ''
+    }
+  }
+
+  @action.bound
+  async initialize(): Promise<void> {
+    try {
+      await this.fetchConfig()
+      await this.fetchFiles()
+      await this.fetchTypings()
+    } catch (err) {
+      console.error('Error while fetching data', err)
+    }
+  }
+
+  @action.bound
+  async fetchConfig() {
+    const config = await this.api.fetchConfig()
+    runInAction('-> setEditorConfig', () => {
+      this.config = config
+    })
+  }
+
+  @action.bound
+  async fetchFiles() {
+    const files = await this.api.fetchFiles()
+    runInAction('-> setFiles', () => {
+      this.files = files
+    })
+  }
+
+  @action.bound
+  async fetchTypings() {
+    const typings = await this.api.fetchTypings()
+    runInAction('-> setTypings', () => {
+      this.typings = typings
+    })
+
+    return this.typings
+  }
+
+  @action.bound
+  setFiles(messages) {
+    this.files = messages
+  }
+
+  @action.bound
+  setFilenameFilter(filter: string) {
+    this.filters.filename = filter
+  }
+
+  @action.bound
+  createFilePrompt(type: FileType, isGlobal?: boolean, hookType?: string) {
```

Comment:
As a personal preference, I always prefer to keep rendering logic in the component (here to display a prompt) and the data logic in the stores